### ISSUE TITLE
No connection PVOutput due to network/DNS timeout/errors

### ISSUE THREAD

**Issue by LudoVicious** [harold65/SolarMeter#2]
At times, my Arduino can not connect to PVOutput. It reports "no connection" on the status page. I found out in the code that both the DNS request and PVOutput network connection are tried only once. If this fails this is interpreted as a PVOutput fail. Therefore I put the DNS request and PVOutput request statements in a loop to retry multiple times (>10) when the connection fails. 
This makes the upload much more reliable in my case. Maybe a thought to be put in a code revision?

**Comment by FunnyAnimal** [comment]
I agree. I replaced the code on line 92 of PVoutput.ino by:

```
int res = 0;
int retries = 5;
while (retries-- > 0 && res != 1)
{
  res = pvout.connect(ip_pvoutput,80);
  busy(0); // Strobe watchdog
}
```

DNS is less of an issue. The IP address is cached, so if dns.getHostByName fails the cached value is used, which is usually fine.

**Comment by harold65** [comment]
Accepted. Will be implemented in 11.44
